cloudify_agent/installer/__init__.py

```diff
@@ -187,7 +194,7 @@ def _create_process_management_options(self):
         # environment variable
         process_management.pop('name')
         for key, value in process_management.iteritems():
-            options.append('--{0}={1}'.format(key, value))
+            options.append("--{0}={1}".format(key, quote(value)))
```

Comment:
This is required because the service password may contain something like a dollar sign. Since the `nssm.conf` file is actually rendered on the manager side, this results in a command-line such as:

`cfy-agent daemons configure ... ... ... --service_password=password$dollar`

Which results in attempting evaluation of `$dollar` as an environment variable.

Long story short... the parameter value must be commandline-quoted.

Reply:
Is it worth raising a separate Jira to make this file be generated as text and dumped into a file rather than rendered via a terminal? (not part of this issue, just a query)

Reply:
@geokala not sure I understand....